### DISCUSSION TITLE
raspberry pi pico async socket not starting

### DISCUSSION THREAD

**Discussion by BxNxM** [micropython/micropython#11984] [RP2040 / Pico] [answered]
I have problem with starting async socket server (while I running async event loop in the background, the code stucks on "        await cls.server". It is only problem on raspberry pi pico w (RP2040). Network mode is AP.

main project "micrOS", server implementation: https://github.com/BxNxM/micrOS/blob/master/micrOS/source/SocketServer.py

"""
   async def run_server(cls):
        """
        Define async socket server (tcp by default)
        """
        addr = ifconfig()[1][0]
        Debug().console(f"[ socket server ] Start socket server on {addr}:{cls.__port}")
        cls.server = asyncio.start_server(cls.handle_client, cls.__host, cls.__port, backlog=cls.__conn_queue)
        await cls.server
        Debug().console(f"- TCP server ready, connect: telnet {addr} {cls.__port}")
"""

If you have any idea, thanks for that :)

**Comment by DavesCodeMusings** [comment, marked as answer]
Have you created an event loop?

For example:
```
from uasyncio import get_event_loop, start_server

loop = get_event_loop()
server = start_server(handler, host, port, queue_len)
loop.create_task(server)
loop.run_forever()
```

**Reply by BxNxM** [reply]
Yes sure, just I have a task manager where I have the following method:

"""
def run_forever(cls):
        try:
            cls.loop.run_forever()
        except Exception as e:
            errlog_add(f"[aio] loop stopped: {e}")
            cls.loop.close()
"""

https://github.com/BxNxM/micrOS/blob/master/micrOS/source/TaskManager.py

cls.loop is the async event loop. But as I mentioned my code running fine on esp32, esp32s2 and s3 as well. So I assume something is flacky in pico w implementation. Maybe :)

**Comment by BxNxM** [comment]
hi, finally I find the issue, btw yes I am starting an event loop :) the host address was empty but it requres 0.0.0.0 then it is working fine :) So my project support raspi pico :) Thanks!
